### DISCUSSION TITLE
Inherited DTO properties appear to be ignored

### DISCUSSION THREAD

**Discussion by Kizzik** [AutoMapper/AutoMapper.Extensions.OData#131] [Q&A] [answered]
I have a series of DTOs with common properties (Id, CreatedOn, CreatedBy) etc that I have split out as a BaseDto. A primitive example:

`public class BaseDto
{   
    public Guid Id { get; set; }
}`

`public class SampleDto : BaseDto
{
    public string Name { get; set; } = null!;
}`

When I retrieve my SampleDtos, the back-end query being generated by EntityFramework looks like this (note that Id isn't included):
`SELECT [s].[Name]
FROM [Samples] AS [s]`

If, however, I have my entity's properties all directly within the DTO, it works as expected:

`public class SampleDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
}`

Will generate:

`SELECT [s].[Id], [s].[Name]
FROM [Samples] AS [s]`

Have I discovered an issue/limitation, or am I doing something wrong?

**Comment by BlaiseD** [comment, marked as answer]
Most likely a bug if that's what you're seeing.  PRs are welcome.

**Reply by BlaiseD** [reply]
This should be resolved now through Issue #132 - thanks for reporting.

**Reply by Kizzik** [reply]
Much appreciated - thank you!
